### PR TITLE
Add support to build linux portable

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @eerhardt @livarcocc @krwq @jonsequitur

### CODE REVIEW

File: run-build.sh [hunk 1]
```diff
@@ -93,6 +96,14 @@ while [[ $# > 0 ]]; do
             args=( "${args[@]/$2}" )
             shift
             ;;
+        --linux-portable)
+            # Pass in a specific version (the one we set up to bootstrap this experience). Once 
+            # we have an official build we can remove the version parameter and use the latest.
+            LINUX_PORTABLE_INSTALL_ARGS="--linux-portable --version 2.0.0-alpha-005151"
```
Comment (eerhardt) [inline run-build.sh]: IMO, it is better to just pin the stage0 to a specific build for all platforms (windows and unix).  That way our build is consistent across everything.

See https://github.com/dotnet/cli/pull/5839 (the non-README changes) for what unpinning stage0 looks like.

File: run-build.sh [hunk 2]
```diff
@@ -93,6 +96,14 @@ while [[ $# > 0 ]]; do
...
+            # Until we get test support for 2.0 we need to pass in the targets without test.
+            LINUX_PORTABLE_BUILD_ARGS="/p:Rid=\"linux-x64\" /p:OSName=\"linux\" /p:IncludeAdditionalSharedFrameworks=false /p:CLITargets=\"Prepare;Compile;Package;Publish\""
+            args=( "${args[@]/$1}" )
+            ;;
         --help)
```
Comment (eerhardt) [inline run-build.sh]: Should probably update the help for `--linux-portable`.

File: run-build.sh [hunk 3]
```diff
@@ -93,6 +96,14 @@ while [[ $# > 0 ]]; do
             args=( "${args[@]/$2}" )
             shift
             ;;
+        --linux-portable)
+            # Pass in a specific version (the one we set up to bootstrap this experience). Once 
+            # we have an official build we can remove the version parameter and use the latest.
+            LINUX_PORTABLE_INSTALL_ARGS="--linux-portable --version 2.0.0-alpha-005151"
+            # Until we get test support for 2.0 we need to pass in the targets without test.
+            LINUX_PORTABLE_BUILD_ARGS="/p:Rid=\"linux-x64\" /p:OSName=\"linux\" /p:IncludeAdditionalSharedFrameworks=false /p:CLITargets=\"Prepare;Compile;Package;Publish\""
```
Comment (eerhardt) [inline run-build.sh]: I don't think `p:IncludeAdditionalSharedFrameworks` should be required to be passed in on the command line.  It should be calculated from the RID, same as what I did here:

https://github.com/dotnet/cli/commit/916a19c4936d92b92b5fbef2b801dba9c125d0db#diff-d595ac18a42f088f16eefcd37b417b04R136

Comment (eerhardt) [inline run-build.sh]: Can we rename `LINUX_PORTABLE_BUILD_ARGS` to be something more general?  That way if other things need to specify build args, they can use the same variable?

File: run-build.sh (hunk 1 above)
Comment (jgoshi) [inline run-build.sh]: Ok will do.

File: run-build.sh (hunk 2 above)
Comment (jgoshi) [inline run-build.sh]: Will do.

File: run-build.sh (hunk 3 above)
Comment (jgoshi) [inline run-build.sh]: Will do. How about CUSTOM_BUILD_ARGS

Comment (eerhardt) [inline run-build.sh]: sounds like a good name to me

**Comment by jgoshi** [comment]
test Ubuntu16.04 x64 Debug Build

**Comment by jgoshi** [comment]
test Ubuntu16.04 x64 Debug Build

**Comment by jgoshi** [comment]
test Ubuntu16.04 x64 Debug Build

File: run-build.sh [hunk 4]
```diff
@@ -102,6 +111,7 @@ while [[ $# > 0 ]]; do
             echo "  --nopackage                         Skip packaging targets"
             echo "  --nobuild                           Skip building, showing the command that would be used to build"
             echo "  --docker <IMAGENAME>                Build in Docker using the Dockerfile located in scripts/docker/IMAGENAME"
+            echo "  --linux-portable                    Builds with the Linux portable .NET Tools instead of a distro-specific version."
```
Comment (eerhardt) [inline run-build.sh]: Shouldn't say "Builds with".  Instead "Builds the Linux portable .NET Tools..."

**Comment by jgoshi** [comment]
test CentOS7.1 x64 Debug Build

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Pull request merged**